File: RP5/sorveteriadener.cpp
#include <iostream> 
#include <vector>
#include <string>

using namespace std;

void exibirCabecalho() {
    cout << "=================================" << endl;
    cout << "     SORVETERIA DO DENER" << endl;
    cout << "   Sistema de Controle de Estoque" << endl;
    cout << "=================================" << endl;
}

string verificarEstoqueCritico(int quantidade) {
    if (quantidade < 5) {
        return "[REPOSIÇÃO NECESSÁRIA]";
    }
    return "";
}

int main() {
    vector<string> nomes;
    vector<int> quantidades;

    exibirCabecalho();

    for (int i = 0; i < 3; i++) {
        string nome;
        int qtd;

        cout << "Digite o nome do insumo: ";
        getline(cin, nome);

        cout << "Digite a quantidade: ";
        cin >> qtd;
        cin.ignore(); // limpa o ENTER do buffer

        nomes.push_back(nome);
        quantidades.push_back(qtd);
    }

    cout << "\nRelatório de Estoque:" << endl;

    for (int i = 0; i < nomes.size(); i++) {
        string aviso = verificarEstoqueCritico(quantidades[i]);
        cout << "Produto: " << nomes[i]
             << " | Estoque: " << quantidades[i]
             << " unidades " << aviso << endl;
    }

    return 0;
}

File: RP5/sorveteriadener.py
def exibir_cabecalho():
    print("=================================")
    print("     SORVETERIA DO DENER ")
    print("   Sistema de Controle de Estoque")
    print("=================================")


def verificar_estoque_critico(quantidade):
    if quantidade < 5:
        return "[REPOSIÇÃO NECESSÁRIA]"
    else:
        return ""


exibir_cabecalho()

nomes = []
quantidades = []

i = 0

while i < 3:
    nome = input("Digite o nome do insumo: ")
    qtd = int(input("Digite a quantidade: "))

    nomes.append(nome)
    quantidades.append(qtd)

    i += 1

print("\nRelatório de Estoque:")

i = 0
while i < len(nomes):

    aviso = verificar_estoque_critico(quantidades[i])

    print(f"Produto: {nomes[i]} | Estoque: {quantidades[i]} unidades {aviso}")

    i += 1